artifact/artifact.cpp

```diff
@@ -0,0 +1,51 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <artifact/artifact.hpp>
+
+#include <common/error.hpp>
+#include <common/expected.hpp>
+
+#include <artifact/error.hpp>
+
+
+namespace mender {
+namespace artifact {
+
+namespace error = mender::common::error;
+namespace expected = mender::common::expected;
+
+
+ExpectedMetaData View(parser::Artifact &artifact, size_t index) {
+	// Check if the inex is available
+	if (artifact.header.info.payloads.size() - 1 < index) {
```

Comment:
For some reason writing it in this order really messed with my mind. I had to look at it for a while until I realized it actually meant: `if (index >= artifact.header.info.payloads.size())`. Does it read better, or is it only me? 🙂

Reply:
haha, lol. True. EOD yesterday, and I was actually staring at it for a minute too :laughing: